ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -490,30 +490,39 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
     ) {
 
       // For now we just support simple prefix/suffix cases only.
-      String paternString = process(node.getPattern(), true).getLeft().substring(1);
-      paternString = paternString.substring(0, paternString.length() - 1);
+      String patternString = process(node.getPattern(), true).getLeft().substring(1);
+      patternString = patternString.substring(0, patternString.length() - 1);
       final String valueString = process(node.getValue(), true).getLeft();
-      if (paternString.startsWith("%")) {
-        if (paternString.endsWith("%")) {
+      if (patternString.startsWith("%")) {
+        if (patternString.endsWith("%")) {
           return new Pair<>(
               "(" + valueString + ").contains(\""
-                  + paternString.substring(1, paternString.length() - 1)
+                  + patternString.substring(1, patternString.length() - 1)
                   + "\")",
               Schema.OPTIONAL_STRING_SCHEMA
           );
         } else {
           return new Pair<>(
-              "(" + valueString + ").endsWith(\"" + paternString.substring(1) + "\")",
+              "(" + valueString + ").endsWith(\"" + patternString.substring(1) + "\")",
               Schema.OPTIONAL_STRING_SCHEMA
           );
         }
       }
 
-      if (paternString.endsWith("%")) {
+      if (patternString.endsWith("%")) {
         return new Pair<>(
             "(" + valueString + ")"
                 + ".startsWith(\""
-                + paternString.substring(0, paternString.length() - 1) + "\")",
+                + patternString.substring(0, patternString.length() - 1) + "\")",
+            Schema.OPTIONAL_STRING_SCHEMA
+        );
+      }
+
+      if (!patternString.contains("%")) {
+        return new Pair<>(
+            "(" + valueString + ")"
+                + ".equals(\""
+                + patternString + "\")",
             Schema.OPTIONAL_STRING_SCHEMA
         );
       }
```

Comment:
Can you add a better error message to the error on line 530: Something like "KSQL only supports leading and trailing wildcards in like expressions".